src/screens/SignIn/SignIn.tsx

```diff
@@ -0,0 +1,225 @@
+/* eslint-disable react-native-a11y/has-valid-accessibility-descriptors */
+import "react-native-gesture-handler";
+import React, { useState } from "react";
+import {
+  View,
+  StyleSheet,
+  Image,
+  Platform,
+  Dimensions,
+  TextInput,
+  SafeAreaView,
+  Pressable,
+  ScrollView,
+} from "react-native";
+import PropTypes from "prop-types";
+import { Button } from "react-native-elements";
+import { useDispatch } from "react-redux";
+import Text from "../../components/Text";
+import { emailSignIn } from "../../firebase/email_signin";
+import { Role } from "../../types";
+import { AuthUser } from "../../redux/reducers/authReducer/types";
+import { login } from "../../redux/reducers/authReducer";
+
+const styles = StyleSheet.create({
+  root: {
+    flex: 1,
+    alignItems: "flex-start",
+    padding: "5%",
+    backgroundColor: "white",
+    flexDirection: "column",
+    width: "100%",
+  },
+  title: {
+    fontSize: 34,
+    textAlign: "center",
+    marginVertical: 8,
+    color: "#4A4B57",
+  },
+  buttonTitle: {
+    alignSelf: "center",
+    fontWeight: "bold",
+    color: "white",
+  },
+  squareButtonTitle: {
+    alignSelf: "center",
+    fontSize: 20,
+    fontWeight: "400",
+  },
+  image: {
+    width: 100,
+    height: Platform.isPad ? 200 : Dimensions.get("window").height * 0.1,
+  },
+  textInput: {
+    height: 55,
+    width: "100%",
+    marginBottom: "5%",
+    borderWidth: 1,
+    borderRadius: 5,
+    borderColor: "#4A4B57",
+    padding: "2%",
+  },
+  textInputTitle: {
+    color: "#4A4B57",
+    fontSize: 14,
+  },
+  errorTitle: {
+    color: "#ed0707",
+    fontSize: 14,
+  },
+});
+
+const logo = require("../../assets/bei.jpg");
+
+//  Home Screen Navigation
+function SignInScreen({ navigation }) {
```

Comment:
Add a type for `navigation` similar to how it's done in screens like `MathMain`, but with reference to the `AuthGuard` stack.

Reply:
Take a look just to make sure I am not missing something. I looked through the React Navigation docs but I am still a bit cautious.